app/code/Magento/Catalog/Block/Adminhtml/Product/Helper/Form/Weight.php

```diff
@@ -84,10 +84,6 @@ public function __construct(
      */
     public function getElementHtml()
     {
-        if (!$this->getForm()->getDataObject()->getTypeInstance()->hasWeight()) {
-            $this->weightSwitcher->setValue(WeightResolver::HAS_NO_WEIGHT);
-        }
-
```

Comment:
I think in this case you need to do the check opposite.

>  if ($this->getForm()->getDataObject()->getTypeInstance()->hasWeight()) {
>             $this->weightSwitcher->setValue(WeightResolver::HAS_WEIGHT);
>         }

Reply:
This check should be

Reply:
You're right @engcom-Echo, sorry for that. I restored this check